code/aepcomposeui/src/main/java/com/adobe/marketing/mobile/aepcomposeui/aepui/components/AepList.kt

```diff
@@ -0,0 +1,139 @@
+import kotlinx.coroutines.flow.MutableStateFlow
+import kotlinx.coroutines.flow.StateFlow
+import kotlinx.coroutines.launch
+
+/**
+ * Composable for rendering a list of AEP UI components.
+ * Maintains a list of AEP UI components and renders them using the provided container.
+ */
+@Composable
+fun AepList(
+    contentProvider: AepUiContentProvider,
+    aepUiEventObserver: AepUiEventObserver,
+    aepUiStyle: AepUiStyle,
+    viewModelKey: String,
+    container: @Composable (@Composable () -> Unit) -> Unit = { content ->
+        // Default to a Column if no container is provided
+        LazyColumn(
+            verticalArrangement = Arrangement.spacedBy(8.dp),
+            horizontalAlignment = Alignment.CenterHorizontally
+        ) {
+            item {
+                content()
+            }
+        }
+    }
+) {
+    val viewModel: AepListViewModel = viewModel(
+        factory = AepComposableViewModelFactory(
+            contentProvider
+        ),
+        key = viewModelKey
+    )
+    val uiList = viewModel.uiList.collectAsStateWithLifecycle()
+
+    container {
+        uiList.value.forEach { ui ->
+            val uiAsComposable = asComposable(ui, aepUiEventObserver, aepUiStyle)
+            uiAsComposable.invoke()
+        }
+
+        Button(onClick = { viewModel.loadMore() }, Modifier.fillMaxWidth()) {
+            Text(text = "Load More")
+        }
+    }
+}
+
+
+private fun asComposable(
+    aepUI: AepUI<*, *>,
+    observer: AepUiEventObserver,
+    aepUiStyle: AepUiStyle
+): @Composable () -> Unit {
+    return when (aepUI) {
+        is SmallImageAepUi -> {
+            {
+                val state = aepUI.getState()
+                if (!state.dismissed) {
+                    SmallImageCard(
+                        ui = aepUI,
+                        style = aepUiStyle.smallImageAepUiStyle,
+                        observer = observer
+                    )
+                }
+            }
+        }
+
+        else -> throw IllegalArgumentException("Unknown template type")
+    }
+}
+
+
+private class AepListViewModel(
+    private val contentProvider: AepUiContentProvider,
+) : ViewModel() {
+
+    private val _uiList = MutableStateFlow(listOf<AepUI<*, *>>())
+    val uiList: StateFlow<List<AepUI<*, *>>> = _uiList
+
+    init {
+        viewModelScope.launch {
+            contentProvider.getContent().collect { templates ->
+                val uiList = templates.map { template ->
+
+                    val aepUiState: AepUI<*, *> = when (template) {
+                        is SmallImageTemplate -> SmallImageAepUi(
+                            template,
+                            SmallImageUIState(title = template.title)
+                        )
+                        else -> throw IllegalArgumentException("Unknown template type")
+                    }
+
+                    aepUiState
+                }
+                _uiList.value = uiList
+            }
+        }
+    }
+
+    fun loadMore() {
```

Comment:
can we call this `refreshList`? The content provider is expected to update the contentFlow when this is called thereby refreshing the list view

Reply:
Also we don't need to support this right now. It was just there for POC. We can add it later if needed